Assets/Chunk.cs

```diff
@@ -226,7 +227,12 @@ private Mesh CreateMesh()
 
     public void SetDirty()
     {
+        if (Dirty || Generating)
+        {
+            return;
+        }
+
         Dirty = true;
-        OnChunkChanged?.Invoke(this, EventArgs.Empty);
+        world.DirtyChunks.Enqueue(this);
```

Comment:
technically if we enqueue dirty chunks the Dirty flag no longer needs to be a thing right?

Reply:
I use the Dirty flag so the chunk doesn't get inserted in the queue multiple times: otherwise the mesh would be updated multiple times, while it maybe wasn't changed.